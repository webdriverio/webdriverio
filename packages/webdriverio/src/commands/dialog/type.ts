/**
 * Returns dialog's type, can be one of `alert`, `beforeunload`, `confirm` or `prompt`.
 *
 * <example>
    :dialogType.js
    const type = await dialog.type();
 * </example>
 *
 * @alias dialog.type
 * @returns {string}  The type of the dialog
 */
// actual implementation is located in packages/webdriverio/src/dialog.ts
