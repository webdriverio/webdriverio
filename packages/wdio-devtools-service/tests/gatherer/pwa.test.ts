import type { CDPSession } from 'puppeteer-core/lib/cjs/puppeteer/common/Connection'
import type { Page } from 'puppeteer-core/lib/cjs/puppeteer/common/Page'

import PWAGatherer from '../../src/gatherer/pwa'

const pageMock = {
    on: jest.fn(),
    url: jest.fn().mockResolvedValue('http://json.org')
} as unknown as Page
const sessionMock = {
    on: jest.fn(),
    send: jest.fn(),
    _connection: { _transport: { _ws: { addEventListener: jest.fn() } } }
} as unknown as CDPSession

describe('PWAGatherer', () => {
    beforeEach(() => {
        (sessionMock.on as jest.Mock).mockClear()
        ;(pageMock.on as jest.Mock).mockClear()
    })

    it('initiates properly', () => {
        const pwaGatherer = new PWAGatherer(sessionMock, pageMock)
        expect((sessionMock.on as jest.Mock).mock.calls.map((c) => c[0]))
            .toMatchSnapshot()
        pwaGatherer['_networkRecorder'].dispatch({ method: 'Network.requestWillBeSent', params: 'bar' })
        expect(pwaGatherer['_networkRecords']).toHaveLength(0)
        expect(pwaGatherer['_networkRecorder'].getRawRecords()).toHaveLength(1)
        ;(pageMock.on as jest.Mock).mock.calls[0][1]()
        expect(pwaGatherer['_networkRecords']).toHaveLength(1)
        expect(pwaGatherer['_networkRecorder'].getRawRecords()).toHaveLength(0)
    })

    it('gatherData', async () => {
        const pwaGatherer = new PWAGatherer(sessionMock, pageMock)
        pwaGatherer['_driver'].getServiceWorkerVersions = jest.fn().mockResolvedValue({ versions: '1.2.1' })
        pwaGatherer['_driver'].getServiceWorkerRegistrations = jest.fn().mockResolvedValue({ registrations: 42 })
        pwaGatherer['_driver'].evaluate = jest.fn().mockResolvedValue({ some: 'result' })
        expect(await pwaGatherer.gatherData()).toMatchSnapshot()
    })
})
