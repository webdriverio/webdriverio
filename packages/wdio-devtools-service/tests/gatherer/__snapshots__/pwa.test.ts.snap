// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PWAGatherer gatherData 1`] = `
Object {
  "InstallabilityErrors": "getInstallabilityErrorsResult",
  "LinkElements": "barfoo",
  "MetaElements": Object {
    "some": "result",
  },
  "ServiceWorker": Object {
    "registrations": 42,
    "versions": "1.2.1",
  },
  "URL": Object {
    "finalUrl": "http://json.org",
    "requestedUrl": "http://json.org",
  },
  "ViewportDimensions": "foobar",
  "WebAppManifest": "getWebAppManifestResult",
}
`;

exports[`PWAGatherer initiates properly 1`] = `
Array [
  "Network.requestWillBeSent",
  "Network.requestServedFromCache",
  "Network.responseReceived",
  "Network.dataReceived",
  "Network.loadingFinished",
  "Network.loadingFailed",
  "Network.resourceChangedPriority",
  "protocolevent",
]
`;
